corehq/pillows/user.py

```diff
@@ -134,6 +134,7 @@ def get_user_pillow(pillow_id='user-pillow', num_processes=1, process_num=0,
     user_processor = get_user_es_processor()
     ucr_processor = ConfigurableReportPillowProcessor(
         data_source_providers=[DynamicDataSourceProvider(), StaticDataSourceProvider()],
```

Comment:
I know there is filtering already happening in the processor but I think it would be nicer to pass filters into the data source provider itself:
```suggestion
        data_source_providers=[
            DynamicDataSourceProvider(referenced_doc_type='CommCareUser'), 
            StaticDataSourceProvider(referenced_doc_type='CommCareUser')
        ],
```

Reply:
I considered, but the underlying view doesn't support the filter anyway, so I added it here. Could do that.

Reply:
Yea I realize that but I do think it makes sense to put filtering in the DSP

Reply:
Updated.

Reply:
@sravfeyn was there a reason you decided not to pass in the doc type to the provider in the constructor?

Reply:
Yeah, I did it that way at first and saw that I had to update the code in all pillows where the providers are instantiated. Whereas with this, I just needed to update the processor code that gets ucr-configs.

Reply:
The kwarg can be optional `referenced_doc_type=None` which means you'd only need to change the ones you're already changing in this PR.

Reply:
It would be all the case/form/location/user pillows and for both static and dynamic ones in each of them. Not a big pref either way, this was much less number of changes

Reply:
that's correct, which are the ones you've already edited in this PR (except for case pillow which I don't see in this PR)

Reply:
Updated.